torch/testing/_internal/common_methods_invocations.py

```diff
@@ -413,8 +413,6 @@ def sample_inputs_batch_norm(op_info, device, dtype, requires_grad, **kwargs):
         ((S, S, S), {'training': True, 'momentum': 0.5, 'eps': 0.6}),
         ((3, 2, 4), {'training': False, 'momentum': -1.2}),
         ((3, 1), {'training': True, 'momentum': 0.0}),
-        ((0,), {'training': True}),
-        ((0,), {'training': False}),
```

Comment:
Why are you removing these test cases?

Reply:
According to the content of this PR modification, batch_norm cannot accept tensors less than 2dims, so these two input data will cause many test cases failed.
I have already add a testcase to cover tensor less than 2dims.

Reply:
The thing is that these two tests used to pass. So if we're saying they're not passing anymore, this would be BC-breaking.
I don't think we should change the behavior of batchnorm in any way here.

We would need to make sure the test only prevent the case that segfault but still allow anything that used to be valid before.

Reply:
Thank you for your reply. But the fact is the only thing I have done in this PR is move input parameters check from outer function to barch_norm itself, because some times, this parameter check is not called(the check is not in every call stack).  I the past, these two testcase can pass because batch_norm can handle empty tensor(it will return another empty tensor as the result), but even so, some code has already visit invalid memory(such as here: https://github.com/pytorch/pytorch/blob/677e67c3998df11bee0d371f9f6f9482c9e4c29e/aten/src/ATen/native/Normalization.cpp#L495, if input only as 1 dim, input.sym_sizes()[1] is not a valid address). That why I create this PR to prevent input less than 2 dims.

For these two test cases, input only has 1 dim, it will checked by new added parameter check, So I remove them. 

Do you have any good suggestions to keep these two test cases but fix this issue? Thanks.

Reply:
Someone would need to dive into details of what is expected or not for the python side of things but the sample just above this one is `((3, 1), {'training': True, 'momentum': 0.0}),` which would mean a 2D input IIRC.

I'm not saying this is bug free but the key expectation for this code is stability. So I do not want to make any change that might change what works or not without understanding in details the state of what is expected.
Sorry for dragging this PR but this is a very high profile function and we definitely don't want to break anything that is working today.

Reply:
Thanks. I got it. I will try to not break anything and fix this issue.